src/algorithms/mul.rs

```diff
@@ -224,7 +224,18 @@ fn mac(lhs: &mut u64, a: u64, b: u64, c: u64) -> u64 {
     prod.high()
 }
 
-/// Computes `lhs += a * b` and returns the borrow.
+/// Computes `lhs *= a` and returns the carry.
+pub fn mul_nx1(lhs: &mut [u64], a: u64) -> u64 {
+    let mut carry = 0;
+    for lhs in lhs.iter_mut() {
```

Comment:
if `lhs` is empty this will return 0. is this intended, or should it return `a`?

Reply:
> my belief is that a 0-byte uint always represents the number 0

This is correct. See for example `U0::ZERO`. (This is also the reason why no `Uint::ONE` can exits)